pkgs/development/libraries/physics/geant4/datasets.nix

```diff
@@ -1,22 +1,25 @@
-{ stdenv, fetchurl, }:
+{ stdenv, fetchurl, pname, pver }:
 
 let
-  mkDataset = { name, version, sha256, envvar}:
+  mkDataset = { name, version, sha256, envvar }:
     stdenv.mkDerivation {
       inherit name version;
+      inherit pname pver;
 
       src = fetchurl {
-        url = "https://geant4-data.web.cern.ch/geant4-data/datasets/${name}.${version}.tar.gz";
+        url = "https://cern.ch/geant4-data/datasets/${name}.${version}.tar.gz";
         inherit sha256;
       };
 
       preferLocalBuild = true;
       dontBuild = true;
       dontConfigure = true;
 
+      # Fix conflict between READMEs of various packages
       installPhase = ''
-        mkdir -p $out/data
-        mv ./* $out/data
+        OUTDIR=$out/share/${pname}-${pver}/data/${name}${version}
```

Comment:
This line can be moved out to the derivation:
```nix
stdenv.mkDerivation {
  # ...
  datadir = "${placeholder "out"}/share/Geant4-${geant_version}/data/${name}${version}";
  installPhase = ''
    mkdir "$datadir"
    mv ./* "$datadir"
  '';
  # ...
}
```
Then it can be used as `@datadir@` in the setup hook, it will be also available for introspection (e.g. `geant4.data.G4NDL.datadir`) from different derivations.
I propose we hardcode the "Geant4" bit. Also could rename `pver` to something more readable like `geant_version`